src/test/java/be/syntra/devshop/DevshopBack/services/utilities/ShopOrderMapperTest.java

```diff
@@ -0,0 +1,77 @@
+package be.syntra.devshop.DevshopBack.services.utilities;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import be.syntra.devshop.DevshopBack.entities.ShopOrder;
+import be.syntra.devshop.DevshopBack.models.CartDto;
+import be.syntra.devshop.DevshopBack.models.ProductDto;
+import be.syntra.devshop.DevshopBack.testutilities.CartUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.List;
+
+import static be.syntra.devshop.DevshopBack.testutilities.ProductUtils.createDummyNonArchivedProductList;
+import static be.syntra.devshop.DevshopBack.testutilities.ProductUtils.createDummyProductDtoList;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+
+public class ShopOrderMapperTest {
+
+    @InjectMocks
+    private CartMapper cartMapper;
+
+    @Mock
+    private ProductMapper productMapper;
+
+    @BeforeEach
+    public void setUp() {
+        initMocks(this);
+    }
+
+    @Test
+    void convertToCartTest() {
+        // given
+        ShopOrder shopOrder = CartUtils.createCart();
+        CartDto cartDto = CartUtils.createCartDto();
+        cartDto.setCartCreationDateTime(shopOrder.getShopOrderCreationDateTime());
+        List<ProductDto> dummyProductDtoList = createDummyProductDtoList();
+        when(productMapper.convertToProductDtoList(any())).thenReturn(dummyProductDtoList);
+
+        // when
+        ShopOrder mappedShopOrder = cartMapper.convertToCart(cartDto);
+
+        // then
+        assertEquals(mappedShopOrder.getClass(), ShopOrder.class);
+        assertEquals(mappedShopOrder.getShopOrderCreationDateTime().getDayOfWeek(), shopOrder.getShopOrderCreationDateTime().getDayOfWeek());
+        assertEquals(mappedShopOrder.getShopOrderCreationDateTime().getHour(), shopOrder.getShopOrderCreationDateTime().getHour());
+        assertEquals(mappedShopOrder.getShopOrderCreationDateTime().getMinute(), shopOrder.getShopOrderCreationDateTime().getMinute());
+        assertEquals(mappedShopOrder.toString(), shopOrder.toString());
+        assertFalse(mappedShopOrder.isFinalizedShopOrder() && cartDto.isFinalizedCart());
```

Comment:
genuinely curious why you'd assertFalse instead of Equals

Reply:
no idea, changed to assertEquals(), can't imagine writing that , it's in the master also, shows as as changed/added due to my refactoring probably, but damn what a weird assertion.....

Reply:
to be clear: pretty sure i didn't write that test originally, still can't wrap my head around the thing we need to assert here and doing it that way.....

Reply:
it's a Thomas original. 'nuff said